### PR TITLE
added cypress and initial test, still working on code coverage and number of tests, but this is a good initial commit with most every configured

### LINKED ISSUE #967: Need more e2e tests
@rtritto there are unwritten tests in https://github.com/mongo-express/mongo-express/blob/master/test/lib/routers/collectionSpec.js

Do you think we can write tests in this "bootstrap" test file ? What is its purpose ? It seems like we could add test data in `testMongoUtils` and write real tests for different queries

_Originally posted by @armellarcier in https://github.com/mongo-express/mongo-express/issues/951#issuecomment-1247286824_

### PR DESCRIPTION
**Pull request by OOCAZ**
This change is [Reviewable](https://reviewable.io/reviews/mongo-express/mongo-express/995)
- - -

Framework for E2E implementation. Will need help making sure the yarn is updated correctly, added several packages.

Fix #967

### CODE REVIEW

**Comment by OOCAZ** [comment]
@BlackthornYugen @rtritto Yarn file probably needs to be updated...

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "livePreview.defaultPreviewPath": "c:\\Users\\zacpo\\Documents\\GitHub\\mongo-express\\coverage\\lcov-report\\index.html"
```
Comment (shakaran) [inline .vscode/settings.json:2]: Why is this file added?

File: cypress.config.cjs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const { defineConfig } = require('cypress')
```
Comment (shakaran) [inline cypress.config.cjs:1]: We are using ESM with ES6, you need to change the require by import

File: cypress/e2e/homepage.cy.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+describe('empty spec', () => {
+  it('passes', () => {
+    cy.visit('localhost:8081');
```
Comment (shakaran) [inline cypress/e2e/homepage.cy.js:3]: Following the .env or NODE_ENV it should load the running local server start, but avoid harcoded values

File: cypress/support/commands.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// ***********************************************
+// This example commands.js shows you how to
```
Comment (shakaran) [inline cypress/support/commands.js:2]: Which give us the example commands? We only should keep the used

File: .vscode/settings.json (hunk 1 above)
Comment (OOCAZ) [reply .vscode/settings.json:2]: I don't know, Can be delted lol

File: cypress.config.cjs (hunk 2 above)
Comment (OOCAZ) [reply cypress.config.cjs:1]: do you know where I need to import require from?

File: cypress/e2e/homepage.cy.js (hunk 3 above)
Comment (OOCAZ) [reply cypress/e2e/homepage.cy.js:3]: Cypress has it's own variable configs, I will work on that and push a fix

File: cypress/support/commands.js (hunk 4 above)
Comment (OOCAZ) [reply cypress/support/commands.js:2]: Understood, Cypress created this and I didn't remove, may be fine to remove

File: cypress.config.cjs (hunk 2 above)
Comment (OOCAZ) [reply cypress.config.cjs:1]: figured it out

File: cypress/support/commands.js (hunk 4 above)
Comment (OOCAZ) [reply cypress/support/commands.js:2]: so this file is required for the E2E testing in the event that a custom command is needed

File: cypress.config.cjs (hunk 2 above)
Comment (rtritto) [reply cypress.config.cjs]: @OOCAZ You must change also the _export_.
It's better to change extension in `.mjs` for ESM ([souce](https://docs.cypress.io/guides/references/configuration#Configuration-File)).

**Comment by shakaran** [comment]
@OOCAZ let me know when you have an update on this and you need rebase with yarn.lock in master

**Comment by OOCAZ** [comment]
> @OOCAZ let me know when you have an update on this and you need rebase with yarn.lock in master

Hoping to get some work done on this later this week. Will tag you here when I need a review.

**Comment by rtritto** [comment]
I can do rebase and the fix `cypress.config` extension file, then I'll test.

**Comment by rtritto** [comment]
Fixed `cypress.config` file, added cypress script and update `.gitignore`.

**Review by shakaran — APPROVED** [review]
